packages/react-core/src/components/NumberInput/__tests__/NumberInput.test.tsx

```diff
@@ -169,4 +169,22 @@ describe('numberInput', () => {
   
     expect(input).toHaveDisplayValue('47.01');
   });
+
+  test('renders 0 if no value passed', () => {
+    const { asFragment } = render(<NumberInput />);
+    const input = screen.getByRole('spinbutton');
+    expect(input).toHaveDisplayValue('0');
+  });
+
+  test('renders 0 if undefined value passed', () => {
+    const { asFragment } = render(<NumberInput value={undefined} />);
+    const input = screen.getByRole('spinbutton');
+    expect(input).toHaveDisplayValue('0');
+  });
+
+  test('renders 0 if null value passed', () => {
+    const { asFragment } = render(<NumberInput value={null} />);
+    const input = screen.getByRole('spinbutton');
+    expect(input).toHaveDisplayValue('0');
+  });
```

Comment:
Nit: can you remove the `asFragment` destructuring since it isn't needed in these tests?

```suggestion
  test('renders 0 if no value passed', () => {
    render(<NumberInput />);
    const input = screen.getByRole('spinbutton');
    expect(input).toHaveDisplayValue('0');
  });

  test('renders 0 if undefined value passed', () => {
    render(<NumberInput value={undefined} />);
    const input = screen.getByRole('spinbutton');
    expect(input).toHaveDisplayValue('0');
  });

  test('renders 0 if null value passed', () => {
    render(<NumberInput value={null} />);
    const input = screen.getByRole('spinbutton');
    expect(input).toHaveDisplayValue('0');
  });
```